lxc/exec_unix.go

```diff
@@ -55,10 +55,17 @@ func (c *cmdExec) controlSocketHandler(control *websocket.Conn) {
 				return
 			}
 		case unix.SIGHUP:
+			file, err := os.OpenFile("/dev/tty", os.O_RDONLY|unix.O_NOCTTY|unix.O_NOFOLLOW|unix.O_CLOEXEC, 0666)
+			if err == nil {
```

Comment:
Can this bit of code be hit when running non-interactively?
If so, we need to change it so we only do that check if in interactive mode.

Also, may be worth adding a check for whether /dev/tty exists, I don't know how standard that is across all Unix platforms, so this change may break macOS for example.

Reply:
> Can this bit of code be hit when running non-interactively?
> If so, we need to change it so we only do that check if in interactive mode.

I mean sure, if you send SIGHUP and you're non-interactive then this would forward SIGTERM.

> 
> Also, may be worth adding a check for whether /dev/tty exists, I don't know how standard that is across all Unix platforms, so this change may break macOS for example.

`/dev/tty` should always be around. It's part of POSIX and a part that most implementations follow.

Reply:
> > Can this bit of code be hit when running non-interactively?
> > If so, we need to change it so we only do that check if in interactive mode.
> 
> I mean sure, if you send SIGHUP and you're non-interactive then this would forward SIGTERM.

But afaict, we don't use the controlsocket handler in that case anyway so signals won't be handled by us and thus this does not apply, no?

Reply:
Indeed, as we don't apply the handler to non-interactive, then we should be fine.
Good to know that POSIX specifies /dev/tty and so we should be fine. If someone complains on macOS/BSD we'll sort it out then.